packages/worker/workflow-isolate-extension.cc

```diff
@@ -27,11 +27,16 @@ void custom_promise_hook(
     Local<Value> parent
 ) {
 #if V8_AT_LEAST(9, 0, 0)
-    Local<Context> context = promise->GetCreationContext().ToLocalChecked();
+    Local<Context> context;
+    bool valid = promise->GetCreationContext().ToLocal(&context);
+    if (!valid) return;
 #else
     Local<Context> context = promise->CreationContext();
 #endif
     Local<Function> fn = Local<Function>::Cast(context->GetEmbedderData(EMBEDDER_DATA_IDX));
+    if (!fn->IsFunction()) {
+        return;
```

Comment:
Should this blow up in a more obvious way?

Reply:
This may happen when creating a promise before registering the promise hook for a context (e.g. when creating the root scope), I'll make sure to add a comment so it's clear.